ddtrace/contrib/internal/django/patch.py

```diff
@@ -535,27 +534,27 @@ def blocked_response():
             )
             core.dispatch("django.start_response.post", ("Django",))
 
-            if core.get_item(HTTP_REQUEST_BLOCKED):
+            if get_blocked():
                 response = blocked_response()
                 return response
 
             try:
                 response = func(*args, **kwargs)
             except BlockingException as e:
-                core.set_item(HTTP_REQUEST_BLOCKED, e.args[0])
+                set_blocked(e.args[0])
                 response = blocked_response()
                 return response
 
-            if core.get_item(HTTP_REQUEST_BLOCKED):
+            if get_blocked():
                 response = blocked_response()
                 return response
 
             return response
         finally:
             core.dispatch("django.finalize_response.pre", (ctx, utils._after_request_tags, request, response))
-            if not core.get_item(HTTP_REQUEST_BLOCKED):
+            if not get_blocked():
                 core.dispatch("django.finalize_response", ("Django",))
-                if core.get_item(HTTP_REQUEST_BLOCKED):
+                if get_blocked():
```

Comment:
How does this work, is this code below not unreachable?

Reply:
dispatching the appsec listener trigger a waf activation that can block the request. So basically here: if we did not already block the request, let's use the WAF (by core.dispatch). OK, and now, maybe we need to block the request.